packages/web-api/src/file-upload.ts

```diff
@@ -227,13 +227,19 @@ Record<string, FilesCompleteUploadExternalArguments> {
     if (file_id) {
       const compareString = `:::${channel_id}:::${thread_ts}:::${initial_comment}`;
       if (!Object.prototype.hasOwnProperty.call(toComplete, compareString)) {
-        toComplete[compareString] = {
-          files: [{ id: file_id, title }],
-          channel_id,
-          initial_comment,
-        };
         if (thread_ts) {
-          toComplete[compareString].thread_ts = upload.thread_ts;
+          toComplete[compareString] = {
+            files: [{ id: file_id, title }],
+            channel_id,
+            initial_comment,
+            thread_ts
+          };
+        }else{
+          toComplete[compareString] = {
+            files: [{ id: file_id, title }],
+            channel_id,
+            initial_comment,
+          };
```

Comment:
toComplete[compareString] has to either be `FileChannelDestinationArgument` or `FileThreadDestinationArgument`

Previously this would make the type dynamically change